UI/LobbyMapUI.cs

```diff
@@ -0,0 +1,899 @@
+            markerComponents.Clear();
+            markerComponents.AddRange(markers.Where(f => lobbyMapInfo.ShouldShowMarker(f)).Select(createMarkerComponent));
+            markerComponents.ForEach(Add);
+            
+            // if this is the first time we've selected a lobby, select the nearest warp
+            if (first && Engine.Scene is Level level && level.Tracker.GetEntity<Player>() is Player player) {
+                selectedWarpIndexes[selectedLobbyIndex] = 0;
+                var nearestWarpLengthSquared = float.MaxValue;
+                for (int i = 0; i < activeWarps.Count; i++) {
+                    var warpPosition = activeWarps[i].Position + level.LevelOffset;
+                    var lengthSquared = (warpPosition - player.Position).LengthSquared();
+                    if (lengthSquared < nearestWarpLengthSquared) {
+                        nearestWarpLengthSquared = lengthSquared;
+                        selectedWarpIndexes[selectedLobbyIndex] = i;
+                    }
+                }
+            }
+
+            var warpIndex = selectedWarpIndexes[selectedLobbyIndex];
+            
+            // get the map texture
+            mapTexture = GFX.Gui[lobbyMapInfo.MapTexture].Texture.Texture;
+
+            // generate the overlay texture
+            overlayTexture?.Dispose();
+            overlayTexture = new Texture2D(Engine.Instance.GraphicsDevice, lobbyMapInfo.RoomWidth, lobbyMapInfo.RoomHeight, false, SurfaceFormat.Alpha8);
+            overlayTexture.SetData(visitedTiles.Data);
+
+            // set view
+            if (zoomLevel < 0) {
+                zoomLevel = defaultZoomLevel;
+            }
+            zoomLevel = Calc.Clamp(zoomLevel, 0, zoomLevels.Length);
+            actualScale = zoomLevels[zoomLevel];
+            shouldCentreOrigin = zoomLevel == 0;
+            selectedOrigin = warpIndex >= 0 && warpIndex < activeWarps.Count ? originForPosition(activeWarps[warpIndex].Position) : new Vector2(0.5f);
+            actualOrigin = shouldCentreOrigin ? new Vector2(0.5f) : selectedOrigin;
+            translateTimeRemaining = 0f;
+            scaleTimeRemaining = 0f;
+
+            // calculate multiplier by aspect ratios
+            var padded = bounds;
+            padded.Inflate(-10, -10);
+            var mapAspectRatio = (float) mapTexture.Width / mapTexture.Height;
+            var boundsAspectRatio = (float) padded.Width / padded.Height;
+            scaleMultiplier = mapAspectRatio > boundsAspectRatio ? (float) padded.Width / mapTexture.Width : (float) padded.Height / mapTexture.Height;
+
+            // update marker positions
+            updateMarkers();
+            
+            // add heart component
+            if (heartSprite != null && !lobbyMapInfo.ShowHeartCount) {
+                heartSprite?.RemoveSelf();
+                heartSprite = null;
+            }
+            else if (lobbyMapInfo.ShowHeartCount) {
+                var heartPath = lobbyMapInfo.LobbyIndex <= 3
```

Comment:
you're still not using `InGameOverworldHelper.GetGuiHeartSpriteId` :(

Reply:
Done, although it still has to fall back in case it can't find a custom sprite.